hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -451,6 +456,37 @@ public abstract AbfsRestOperation createPath(final String path,
       final ContextEncryptionAdapter contextEncryptionAdapter,
       final TracingContext tracingContext, boolean isNamespaceEnabled) throws AzureBlobFileSystemException;
 
+  public AbfsRestOperation createNonRecursivePath(final String pathStr,
+      final boolean isFile,
+      final boolean overwrite,
+      final Permissions permissions,
+      final boolean isAppendBlob,
+      final String eTag,
+      final ContextEncryptionAdapter contextEncryptionAdapter,
+      final TracingContext tracingContext, boolean isNamespaceEnabled)
+      throws IOException {
+    Path parentPath = new Path(pathStr).getParent();
+
+    try {
+      getPathStatus(parentPath.toString(), false, tracingContext,
+          contextEncryptionAdapter);
+    } catch (AbfsRestOperationException ex) {
+      if (ex.getStatusCode() == HttpURLConnection.HTTP_OK) {
+        throw new FileNotFoundException("Cannot create file "
+            + pathStr + " because parent folder does not exist.");
+      }
+      throw ex;
+    } finally {
+      abfsCounters.incrementCounter(CALL_GET_FILE_STATUS, 1);
```

Comment:
And why getFileStatus increment ?

Reply:
GetFileStatus API is getting called, and there is a test that asserts it. Hence added here.

Reply:
I see only a GetPathStatus API call

Reply:
yes, its for that, in trunk, it calls tryGetFileStatus -> getFileStatus -> store.getFileStatus -> client.getPathStatus